### PR TITLE
[flutter release] Add cherry pick template for pull request description

### PR DESCRIPTION
**Pull request by XilaiZhang**
context:  https://github.com/flutter/flutter/pull/139524#discussion_r1416052456

Adds a pull request template for automated flutter cherry picks.
This template is adapted from  [web form cherry pick issue template](https://cs.opensource.google/flutter/flutter/+/master:.github/ISSUE_TEMPLATE/7_cherry_pick.yml) since PR template doesn't support web form.

This would provide the template to allow https://github.com/flutter/flutter/pull/139524 to land

### CODE REVIEW

**Review by CaseyHillers — CHANGES_REQUESTED** [review]
Instead of creating duplicate templates, let's reuse https://github.com/flutter/flutter/blob/master/.github/ISSUE_TEMPLATE/7_cherry_pick.yml. As we migrate off of creating issues, and using the PRs directly, we can clean up the template.

**Comment by XilaiZhang** [comment]
Right the current template had to be adapted from the web form issue template(https://github.com/flutter/flutter/blob/master/.github/ISSUE_TEMPLATE/7_cherry_pick.yml).

The pull request template only supports markdown, and a direct copy and paste of web form issue template would cause texts to be corrupted (https://github.com/flutter/flutter/pull/139405)

The current PR template is in markdown format so that it can be displayed correctly in a PR description (e.g. https://github.com/flutter/flutter/pull/139594)

**Comment by CaseyHillers** [comment]
I defer to Kevin. I lean towards us updating the template so it can support both use cases, but he has more context on the how.

**Comment by itsjustkevin** [comment]
@XilaiZhang is the point of this template to be directly consumed by the github action you are creating and used in the body of the automated cherry-pick request?

If that is the case, we should prefer to create the template in a separate folder inside of .github to avoid confusion and separate concerns.

@CaseyHillers Xilai is correct in stating that we can not reuse the current issue template for `cherry_pick.yml`.  YAML form syntax is only available for issues and will not work with PRs.  Please let me know if I am misunderstanding your comment.

**Comment by CaseyHillers** [comment]
> @CaseyHillers Xilai is correct in stating that we can not reuse the current issue template for cherry_pick.yml. YAML form syntax is only available for issues and will not work with PRs. Please let me know if I am misunderstanding your comment.

That makes sense to me, I'm more interested if we should convert this to a markdown template so we can share them between CP issues and PRs. I think it's fine to keep both as well.

File: .github/PULL_REQUEST_CP_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+**This pull request is opened against a release branch.<br>
+To request a cherry pick of a commit, please fill in the form below.** (Questions with an asterisk are required.)<br>
+
+**To fill in the form, you can edit this PR description and type your answers after the 'My Answer' keywords. <br>
+A flutter domain expert will evaluate this cherry pick request shortly after ALL questions are answered.**
+
+* Changelog Description: Explain this cherry pick in one line that is accessible to most Flutter developers
```
Comment (CaseyHillers) [inline .github/PULL_REQUEST_CP_TEMPLATE.md:7]: This is missing the section with the original issue

**Comment by itsjustkevin** [comment]
> That makes sense to me, I'm more interested if we should convert this to a markdown template so we can share them between CP issues and PRs. I think it's fine to keep both as well.

The community has already done the work of converting most of our markdown issue templates to the new YAML format.  

There is no news, but I am holding out hope that PR templates can use the YAML format in the future.  

I think we should stick to the YAML format where possible.

File: .github/PULL_REQUEST_CP_TEMPLATE.md (hunk 1 above)
Comment (XilaiZhang) [reply .github/PULL_REQUEST_CP_TEMPLATE.md]: Sure, added the original issue field.

Also added text box styling around the answers fields, it now looks like the PR description of https://github.com/flutter/flutter/pull/139594.

I looked into [colored blocks](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) but looks like they are for alerts and not regular text blocks.

**Comment by XilaiZhang** [comment]
> If that is the case, we should prefer to create the template in a separate folder inside of .github to avoid confusion and separate concerns.

Sure, moved to a separate folder

**Review by itsjustkevin — APPROVED** [review]

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**